## Final Project Submission

Please fill out:
* Student name: 
* Student pace: self paced / part time / full time
* Scheduled project review date/time: 
* Instructor name: 
* Blog post URL:


In [ ]:
# Your code here - remember to use markdown cells for comments as well!

The Scenario

The company you work for is interested in purchasing and operating airplanes for commercial and private enterprises, but know nothing about the potential risks of aircraft.
You are charged with determining which aircraft are the lowest risk for the company to start this new business endeavor.

The Data

aviation accident data from 1962 to 2023 about civil aviation accidents 
up to you to decide:
-> what data to use
-> how to deal with missing values
-> how to aggregate (ie. sum, mean, median, count, max..) the data
-> how to visualize it in an interactive dashboard

The Expectation

make 3 recommendations for how the business should move forward
insights to help the head of the new aviation division decide which aircraft to purchase

You should make:

A non-technical presentation (ie. slides)
A Jupyter Notebook
A GitHub repository
An Interactive Dashboard (ie. tableau)

In [135]:
import pandas as pd

In [136]:
aviation_df = pd.read_csv('Aviation_Data.csv')
aviation_df

c:\Users\User\anaconda3\envs\learn-env\lib\site-packages\IPython\core\interactiveshell.py:3145: DtypeWarning: Columns (6,7,28) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country Latitude Longitude Airport.Code  \
0      MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1       BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2        Saltville, VA  United States  36.9222  -81.8781          NaN   
3           EUREKA, CA  United States      NaN       NaN          NaN   
4           Canton, OH  United States      NaN       NaN          NaN   
...                ...            ...      ...       ...          ...   
90343    Annapolis, MD  United States      NaN       NaN          NaN   
90344      Hampton, NH  United States      NaN       NaN          NaN   
90345       Payson, AZ  United States  341525N  1112021W          PAN   
90346       Morgan, UT  United States      NaN       NaN          NaN   
90347       Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
0              NaN  ...          Personal                 NaN   
1              NaN  ...          Personal                 NaN   
2              NaN  ...          Personal                 NaN   
3              NaN  ...          Personal                 NaN   
4              NaN  ...          Personal                 NaN   
...            ...  ...               ...                 ...   
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                      2.0                    0.0                  0.0   
1                      4.0                    0.0                  0.0   
2                      3.0                    NaN                  NaN   
3                      2.0                    0.0                  0.0   
4                      1.0                    2.0                  NaN   
...                    ...                    ...                  ...   
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight  \
0                 0.0               UNK                 Cruise   
1                 0.0               UNK                Unknown   
2                 NaN               IMC                 Cruise   
3                 0.0               IMC                 Cruise   
4                 0.0               VMC               Approach   
...               ...               ...                    ...   
90343             0.0               NaN                    NaN   
90344             0.0               NaN                    NaN   
90345  

some eda

In [137]:
aviation_df.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

In [138]:
aviation_df['Number.of.Engines'].value_counts()


1.0    69582
2.0    11079
0.0     1226
3.0      483
4.0      431
8.0        3
6.0        1
Name: Number.of.Engines, dtype: int64

In [139]:
aviation_df.iloc[0:10, [14,15,23,24,25,27]]


Make         Model  Total.Fatal.Injuries  \
0            Stinson         108-3                   2.0   
1              Piper      PA24-180                   4.0   
2             Cessna          172M                   3.0   
3           Rockwell           112                   2.0   
4             Cessna           501                   1.0   
5  Mcdonnell Douglas           DC9                   NaN   
6             Cessna           180                   4.0   
7             Cessna           140                   0.0   
8             Cessna          401B                   0.0   
9     North American  NAVION L-17B                   0.0   

   Total.Serious.Injuries  Total.Minor.Injuries Weather.Condition  
0                     0.0                   0.0               UNK  
1                     0.0                   0.0               UNK  
2                     NaN                   NaN               IMC  
3                     0.0                   0.0               IMC  
4                     2.0                   NaN               VMC  
5                     NaN                   1.0               VMC  
6                     0.0                   0.0               IMC  
7                     0.0                   0.0               VMC  
8                     0.0                   0.0               IMC  
9                     0.0                   3.0               IMC

In [140]:
aviation_df['Aircraft.damage'].value_counts()['Destroyed']


18623

In [141]:
aviation_df['Investigation.Type'].value_counts()['Accident']

85015

In [142]:
aviation_df['Model'].value_counts()['747-4F6B']

1

In [200]:
aviation_df['Investigation.Type'].value_counts()

Accident      85015
Incident       3874
25-09-2020      702
26-09-2020       60
02-02-2021       39
              ...  
04-08-2021        1
18-11-2020        1
06-01-2021        1
05-04-2021        1
05-08-2022        1
Name: Investigation.Type, Length: 71, dtype: int64

In [144]:
aviation_df['Total.Fatal.Injuries'].isnull().sum()


12860

In [145]:
aviation_df['Make'] = aviation_df['Make'].str.upper()
aviation_df['Model'] = aviation_df['Model'].str.upper()


In [146]:
aviation_df['Model.Count'] = aviation_df['Model'].map(aviation_df['Model'].value_counts())


will impute injury/fatality NaNs with 0

In [147]:
aviation_df['Total.Fatal.Injuries'] = aviation_df['Total.Fatal.Injuries'].fillna(0)
aviation_df['Total.Serious.Injuries'] = aviation_df['Total.Fatal.Injuries'].fillna(0)
aviation_df['Total.Minor.Injuries'] = aviation_df['Total.Minor.Injuries'].fillna(0)
aviation_df['Total.Uninjured'] = aviation_df['Total.Uninjured'].fillna(0)


now confirming the clean

In [148]:
aviation_df['Total.Fatal.Injuries'].isnull().sum()


0

In [149]:
aviation_df['Total.Serious.Injuries'].isnull().sum()


0

In [150]:
aviation_df['Total.Minor.Injuries'].isnull().sum()

0

In [151]:
aviation_df['Total.Uninjured'].isnull().sum()

0

attempting to define a 'severity index' column. giving different 'weight' to death vs serious injury vs minor injury

In [152]:
aviation_df['Severity.Index'] = (
    aviation_df['Total.Fatal.Injuries'] * 5 + 
    aviation_df['Total.Serious.Injuries'] * 3 + 
    aviation_df['Total.Minor.Injuries'] * 1
)

In [153]:
aviation_df.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date', 'Model.Count', 'Severity.Index'],
      dtype='object')

creating 7 new dfs

In [154]:
avg_severity = aviation_df.groupby(['Make','Model'])['Severity.Index'].mean().sort_values(ascending=False)
avg_severity_df = avg_severity.reset_index(name='per.Accident.Severity')


observations_df = (aviation_df.groupby(['Make','Model']).size().reset_index(name='Aircraft.Observations').sort_values('Aircraft.Observations', ascending=False))


accidents_df = (aviation_df[aviation_df['Investigation.Type'] == 'Accident'].groupby(['Make', 'Model']).size().reset_index(name='Aircraft.Accidents')).sort_values('Aircraft.Accidents', ascending=False)


avg_craft_deaths = aviation_df.groupby(['Make','Model'])['Total.Fatal.Injuries'].mean().sort_values(ascending=False)
avg_craft_deaths_df = avg_craft_deaths.reset_index(name='per.Accident.Deaths')


avg_craft_serious = aviation_df.groupby(['Make','Model'])['Total.Serious.Injuries'].mean().sort_values(ascending=False)
avg_craft_serious_df = avg_craft_serious.reset_index(name='per.A.Serious.Injuries')


avg_craft_minor = aviation_df.groupby(['Make','Model'])['Total.Minor.Injuries'].mean().sort_values(ascending=False)
avg_craft_minor_df = avg_craft_minor.reset_index(name='per.A.Minor.Injuries')


avg_craft_uninjured = aviation_df.groupby(['Make','Model'])['Total.Uninjured'].mean().sort_values(ascending=False)
avg_craft_uninjured_df = avg_craft_uninjured.reset_index(name='per.Accident.Uninjured')


In [155]:
avg_severity_df

Make               Model  per.Accident.Severity
0                 TUPOLEV              TU-154                 2792.0
1                  BOEING             747-168                 2792.0
2                  BOEING          767-366-ER                 1736.0
3                  BOEING           777 - 206                 1424.0
4       MCDONNELL DOUGLAS             DC-8-62                 1392.0
...                   ...                 ...                    ...
18241              JAVRON       PA-18 REPLICA                    0.0
18242              JASPER      THUNDERBIRD-2C                    0.0
18243             JAQUISH  CHALLENGER SPEC II                    0.0
18244  JANNAKOS GREGORY P      ZODIAC 601 HDS                    0.0
18245              HAWKER     SEA FURY TMK 20                    0.0

[18246 rows x 3 columns]

In [156]:
observations_df

Make        Model  Aircraft.Observations
4610                   CESSNA          152                   2366
4634                   CESSNA          172                   1753
4685                   CESSNA         172N                   1163
13404                   PIPER    PA-28-140                    932
4583                   CESSNA          150                    829
...                       ...          ...                    ...
7468   EUROCOPTER DEUTSCHLAND      BK117C1                      1
7469   EUROCOPTER DEUTSCHLAND       BO-105                      1
7470   EUROCOPTER DEUTSCHLAND  BO-105 CBS5                      1
7471   EUROCOPTER DEUTSCHLAND    EC 135 P2                      1
18245        ZWICKER MURRAY R      GLASTAR                      1

[18246 rows x 3 columns]

In [157]:
accidents_df

Make      Model  Aircraft.Accidents
4219                           CESSNA        152                2348
4243                           CESSNA        172                1744
4294                           CESSNA       172N                1160
12772                           PIPER  PA-28-140                 929
4192                           CESSNA        150                 822
...                               ...        ...                 ...
7111                EXTRA FLUGZEUGBAU   GMBH 300                   1
7114   EXTRA FLUGZEUGPRODUKTIONS-GMBH   EA-300/L                   1
7115    EXTRA FLUGZEUGPRODUKTIONS-UND     EA 300                   1
7117           EXTRA FLUGZEUGRAU GMBH   EA 300/L                   1
17536                ZWICKER MURRAY R    GLASTAR                   1

[17537 rows x 3 columns]

In [158]:
avg_craft_deaths_df

Make                 Model  per.Accident.Deaths
0                      TUPOLEV                TU-154                349.0
1                       BOEING               747-168                349.0
2                       BOEING            767-366-ER                217.0
3                       BOEING             777 - 206                178.0
4            MCDONNELL DOUGLAS               DC-8-62                174.0
...                        ...                   ...                  ...
18241           LIMING LOUIS L  HPL 1 HIGH WING PARA                  0.0
18242                    LIMES         CHALLENGER II                  0.0
18243                     LIMA          SKY RAIDER I                  0.0
18244  LIGHTNING AVION EAB LLC       ARION LIGHTNING                  0.0
18245                   HAWKER       SEA FURY TMK 20                  0.0

[18246 rows x 3 columns]

In [159]:
avg_craft_serious_df

Make                 Model  per.A.Serious.Injuries
0                      TUPOLEV                TU-154                   349.0
1                       BOEING               747-168                   349.0
2                       BOEING            767-366-ER                   217.0
3                       BOEING             777 - 206                   178.0
4            MCDONNELL DOUGLAS               DC-8-62                   174.0
...                        ...                   ...                     ...
18241           LIMING LOUIS L  HPL 1 HIGH WING PARA                     0.0
18242                    LIMES         CHALLENGER II                     0.0
18243                     LIMA          SKY RAIDER I                     0.0
18244  LIGHTNING AVION EAB LLC       ARION LIGHTNING                     0.0
18245                   HAWKER       SEA FURY TMK 20                     0.0

[18246 rows x 3 columns]

In [160]:
avg_craft_minor_df

Make                 Model  per.A.Minor.Injuries
0                        BOEING              747-2B6B                 380.0
1                        BOEING               737-297                  57.0
2              AIRBUS INDUSTRIE              A340-300                  43.0
3                        BOEING               747-256                  35.0
4                     COMMANDER                 112TC                  30.0
...                         ...                   ...                   ...
18241                    MARTIN  STODDARD HAMILTON II                   0.0
18242                    MARTIN    SORRELL HIPERLIGHT                   0.0
18243                    MARTIN                 SONEX                   0.0
18244                    MARTIN      RANS S-7 COURIER                   0.0
18245  107.5 FLYING CORPORATION     ONE DESIGN DR 107                   0.0

[18246 rows x 3 columns]

In [161]:
avg_craft_uninjured_df

Make              Model  per.Accident.Uninjured
0                        BOEING           747-4F6B                   699.0
1                        BOEING             747-SP                   699.0
2             MCDONNELL DOUGLAS              MD 83                   501.0
3                        BOEING            747-130                   471.5
4                        BOEING           747-243B                   459.0
...                         ...                ...                     ...
18241                   DOUGLAS            C54A-DC                     0.0
18242                     PIPER        PA-22-108/U                     0.0
18243                     PIPER          PA-20-180                     0.0
18244                   DOUGLAS       DC-3C/BT-67R                     0.0
18245  107.5 FLYING CORPORATION  ONE DESIGN DR 107                     0.0

[18246 rows x 3 columns]

merging the new dfs, then sorting in order of avg aircraft risk

In [165]:
merged_frame = avg_craft_uninjured_df.merge(avg_craft_minor_df, on=['Make','Model'], how='inner')
merged_frame = merged_frame.merge(avg_craft_serious_df, on=['Make','Model'], how='inner')
merged_frame = merged_frame.merge(avg_craft_deaths_df, on=['Make','Model'], how='inner')
merged_frame = merged_frame.merge(accidents_df, on=['Make','Model'], how='inner')
merged_frame = merged_frame.merge(observations_df, on=['Make','Model'], how='inner')
merged_frame = merged_frame.merge(avg_severity_df, on=['Make','Model'], how='inner')


In [166]:
merged_frame.sort_values('per.Accident.Severity', ascending=False).head(10)


Make        Model  per.Accident.Uninjured  \
10115             BOEING      747-168                     0.0   
12528            TUPOLEV       TU-154                     0.0   
10708             BOEING   767-366-ER                     0.0   
10715             BOEING    777 - 206                     0.0   
14918  MCDONNELL DOUGLAS      DC-8-62                     0.0   
11183   AIRBUS INDUSTRIE        A 310                     0.0   
11075             AIRBUS   A320 - 216                     0.0   
708              EMBRAER  E135 LEGACY                     7.0   
10622             BOEING        MD-83                     0.0   
534     AIRBUS INDUSTRIE     A310-300                    17.0   

       per.A.Minor.Injuries  per.A.Serious.Injuries  per.Accident.Deaths  \
10115                   0.0                   349.0                349.0   
12528                   0.0                   349.0                349.0   
10708                   0.0                   217.0                217.0   
10715                   0.0                   178.0                178.0   
14918                   0.0                   174.0                174.0   
11183                  10.0                   169.0                169.0   
11075                   0.0                   162.0                162.0   
708                     0.0                   154.0                154.0   
10622                   0.0                   153.0                153.0   
534                     0.0                   124.0                124.0   

       Aircraft.Accidents  Aircraft.Observations  per.Accident.Severity  
10115                   1                      1                 2792.0  
12528                   1                      1                 2792.0  
10708                   1                      1                 1736.0  
10715                   3                      3                 1424.0  
14918                   1                      1                 1392.0  
11183                   1                      1                 1362.0  
11075                   1                      1                 1296.0  
708                     1                      1                 1232.0  
10622                   1                      1                 1224.0  
534                     1                      1                  992.0

for many of these planes, the severity of the accidents is very high, though the accidents themselves are very rare

this signals to me that for a particular make & model, high accident count doesn’t necessarily mean high avg-severity, and high avg-severity doesn't necessarily mean high accident count

adding new columns to the merged df, then sorting in order of total severity

In [167]:
merged_frame['Total.Severity'] = merged_frame['per.Accident.Severity'] * merged_frame['Aircraft.Accidents']
merged_frame['Total.Aircraft.Deaths'] = merged_frame['per.Accident.Deaths'] * merged_frame['Aircraft.Accidents']
merged_frame['Accident.Rate'] = merged_frame['Aircraft.Accidents'] / merged_frame['Aircraft.Observations']
merged_frame['Mortality.Rate'] = merged_frame['Total.Aircraft.Deaths'] / ((merged_frame['per.Accident.Uninjured'] * merged_frame['Aircraft.Accidents']) + (merged_frame['per.A.Minor.Injuries'] * merged_frame['Aircraft.Accidents']) + (merged_frame['per.A.Serious.Injuries'] * merged_frame['Aircraft.Accidents']) + merged_frame['Total.Aircraft.Deaths'])
merged_frame['Survival.Rate'] = 1 - merged_frame['Mortality.Rate']

In [168]:
merged_frame

Make              Model  per.Accident.Uninjured  \
0                        BOEING           747-4F6B                   699.0   
1                        BOEING             747-SP                   699.0   
2             MCDONNELL DOUGLAS              MD 83                   501.0   
3                        BOEING           747-243B                   459.0   
4                        BOEING            777-281                   450.0   
...                         ...                ...                     ...   
17532                   DOUGLAS            C54A-DC                     0.0   
17533                     PIPER        PA-22-108/U                     0.0   
17534                     PIPER          PA-20-180                     0.0   
17535                   DOUGLAS       DC-3C/BT-67R                     0.0   
17536  107.5 FLYING CORPORATION  ONE DESIGN DR 107                     0.0   

       per.A.Minor.Injuries  per.A.Serious.Injuries  per.Accident.Deaths  \
0                       0.0                     0.0                  0.0   
1                       0.0                     0.0                  0.0   
2                       0.0                     0.0                  0.0   
3                       0.0                     0.0                  0.0   
4                       0.0                     0.0                  0.0   
...                     ...                     ...                  ...   
17532                   0.0                     2.0                  2.0   
17533                   2.0                     0.0                  0.0   
17534                   0.0                     0.0                  0.0   
17535                   0.0                     4.0                  4.0   
17536                   0.0                     1.0                  1.0   

       Aircraft.Accidents  Aircraft.Observations  per.Accident.Severity  \
0                       1                      1                    0.0   
1                       1                      1                    0.0   
2                       1                      1                    0.0   
3                       1                      1                    0.0   
4                       1                      1                    0.0   
...                   ...                    ...                    ...   
17532                   1                      1                   16.0   
17533                   1                      1                    2.0   
17534                   1                      1                    0.0   
17535                   1                      1                   32.0   
17536                   1                      1                    8.0   

       Total.Severity  Total.Aircraft.Deaths  Accident.Rate  Mortality.Rate  \
0                 0.0                    0.0            1.0             0.0   
1                 0.0                    0.0            1.0             0.0   
2                 0.0                    0.0            1.0             0.0   
3                 0.0                    0.0            1.0             0.0   
4                 0.0                    0.0            1.0             0.0   
...               ...                    ...            ...             ...   
17532            16.0                    2.0            1.0             0.5   
17533             2.0                    0.0            1.0             0.0   
17534             0.0                    0.0            1.0             NaN   
17535            32.0                    4.0            1.0             0.5   
17536             8.0                    1.0            1.0             0.5   

       Survival.Rate  
0                1.0  
1                1.0  
2                1.0  
3                1.0  
4                1.0  
...              ...  
17532            0.5  
17533            1.0  
17534            NaN  
17535            0.5  
17536            0.5  

[17537 rows x 14 columns]

In [169]:
merged_frame.sort_values('Total.Severity', ascending=False).head(10)


Make      Model  per.Accident.Uninjured  per.A.Minor.Injuries  \
381     BOEING    737-200               50.283019              1.320755   
10715   BOEING  777 - 206                0.000000              0.000000   
4478    CESSNA       172N                1.160791              0.293207   
4290    CESSNA        172                1.283514              0.223046   
8724    CESSNA        152                0.999155              0.177937   
4393     PIPER  PA-28-181                1.215501              0.347826   
374     BOEING        737               52.026585              0.186094   
8808     PIPER  PA-28-140                0.905579              0.436695   
12528  TUPOLEV     TU-154                0.000000              0.000000   
10115   BOEING    747-168                0.000000              0.000000   

       per.A.Serious.Injuries  per.Accident.Deaths  Aircraft.Accidents  \
381                 17.094340            17.094340                  35   
10715              178.000000           178.000000                   3   
4478                 0.368014             0.368014                1160   
4290                 0.227039             0.227039                1744   
8724                 0.164835             0.164835                2348   
4393                 0.718336             0.718336                 525   
374                  2.756646             2.756646                 135   
8808                 0.344421             0.344421                 929   
12528              349.000000           349.000000                   1   
10115              349.000000           349.000000                   1   

       Aircraft.Observations  per.Accident.Severity  Total.Severity  \
381                       53             138.075472     4832.641509   
10715                      3            1424.000000     4272.000000   
4478                    1163               3.237317     3755.288048   
4290                    1753               2.039361     3556.645750   
8724                    2366               1.496619     3514.060862   
4393                     529               6.094518     3199.621928   
374                      489              22.239264     3002.300613   
8808                     932               3.192060     2965.423820   
12528                      1            2792.000000     2792.000000   
10115                      1            2792.000000     2792.000000   

       Total.Aircraft.Deaths  Accident.Rate  Mortality.Rate  Survival.Rate  
381               598.301887       0.660377        0.199252       0.800748  
10715             534.000000       1.000000        0.500000       0.500000  
4478              426.895959       0.997420        0.168041       0.831959  
4290              395.956646       0.994866        0.115799       0.884201  
8724              387.032967       0.992392        0.109397       0.890603  
4393              377.126654       0.992439        0.239445       0.760555  
374               372.147239       0.276074        0.047754       0.952246  
8808              319.966738       0.996781        0.169572       0.830428  
12528             349.000000       1.000000        0.500000       0.500000  
10115             349.000000       1.000000        0.500000       0.500000

the plane with the most fatalities had 35 accidents, and an accident rate of 0.66. numbers 3-5 had 1160, 1744 and 2348 respectively, with accident rates of 0.99
so the total severity from accidents for planes with low average severity adds up if they crash frequently enough

In [171]:
merged_frame.columns

Index(['Make', 'Model', 'per.Accident.Uninjured', 'per.A.Minor.Injuries',
       'per.A.Serious.Injuries', 'per.Accident.Deaths', 'Aircraft.Accidents',
       'Aircraft.Observations', 'per.Accident.Severity', 'Total.Severity',
       'Total.Aircraft.Deaths', 'Accident.Rate', 'Mortality.Rate',
       'Survival.Rate'],
      dtype='object')

In [175]:
merged_frame.corr()

per.Accident.Uninjured  per.A.Minor.Injuries  \
per.Accident.Uninjured                1.000000              0.234622   
per.A.Minor.Injuries                  0.234622              1.000000   
per.A.Serious.Injuries                0.016673              0.016264   
per.Accident.Deaths                   0.016673              0.016264   
Aircraft.Accidents                   -0.007043             -0.003190   
Aircraft.Observations                 0.002043             -0.001223   
per.Accident.Severity                 0.022021              0.039093   
Total.Severity                        0.040957              0.032918   
Total.Aircraft.Deaths                 0.037886              0.020303   
Accident.Rate                        -0.373401             -0.092705   
Mortality.Rate                       -0.115768             -0.143672   
Survival.Rate                         0.115768              0.143672   

                        per.A.Serious.Injuries  per.Accident.Deaths  \
per.Accident.Uninjured                0.016673             0.016673   
per.A.Minor.Injuries                  0.016264             0.016264   
per.A.Serious.Injuries                1.000000             1.000000   
per.Accident.Deaths                   1.000000             1.000000   
Aircraft.Accidents                   -0.004594            -0.004594   
Aircraft.Observations                -0.003497            -0.003497   
per.Accident.Severity                 0.999739             0.999739   
Total.Severity                        0.468237             0.468237   
Total.Aircraft.Deaths                 0.491546             0.491546   
Accident.Rate                        -0.039951            -0.039951   
Mortality.Rate                        0.176527             0.176527   
Survival.Rate                        -0.176527            -0.176527   

                        Aircraft.Accidents  Aircraft.Observations  \
per.Accident.Uninjured           -0.007043               0.002043   
per.A.Minor.Injuries             -0.003190              -0.001223   
per.A.Serious.Injuries           -0.004594              -0.003497   
per.Accident.Deaths              -0.004594              -0.003497   
Aircraft.Accidents                1.000000               0.996470   
Aircraft.Observations             0.996470               1.000000   
per.Accident.Severity            -0.004664              -0.003522   
Total.Severity                    0.668816               0.683321   
Total.Aircraft.Deaths             0.635458               0.650754   
Accident.Rate                    -0.007285              -0.031284   
Mortality.Rate                    0.006818               0.005301   
Survival.Rate                    -0.006818              -0.005301   

                        per.Accident.Severity  Total.Severity  \
per.Accident.Uninjured               0.022021        0.040957   
per.A.Minor.Injuries                 0.039093        0.032918   
per.A.Serious.Injuries               0.999739        0.468237   
per.Accident.Deaths                  0.999739        0.468237   
Aircraft.Accidents                  -0.004664        0.668816   
Aircraft.Observations               -0.003522        0.683321   
per.Accident.Severity                1.000000        0.468693   
Total.Severity                       0.468693        1.000000   
Total.Aircraft.Deaths                0.491699        0.998553   
Accident.Rate                       -0.042043       -0.094275   
Mortality.Rate                       0.173160        0.104720   
Survival.Rate                       -0.173160       -0.104720   

                        Total.Aircraft.Deaths  Accident.Rate  Mortality.Rate  \
per.Accident.Uninjured               0.037886      -0.373401       -0.115768   
per.A.Minor.Injuries                 0.020303      -0.092705       -0.143672   
per.A.Serious.Injuries               0.491546      -0.039951        0.176527   
per.Accident.Deaths                  0.491546      -0.039951        0.176527   
Aircraft.Accidents 

creating a crash_survival df that filters the merged df for a threshold number of accidents, people uninjured per accident and total aircraft deaths for each aircraft

In [204]:
filtered_survival = merged_frame.loc[
    (merged_frame['Aircraft.Accidents'] >= 20) &
    (merged_frame['Total.Aircraft.Deaths'] <= 2)
]

filtered_survival.sort_values('Survival.Rate', ascending=False)



Make          Model  per.Accident.Uninjured  \
251                           BOEING            777               99.147368   
4162                          HILLER         UH-12B                1.363636   
4613              AVIAT AIRCRAFT INC           A-1B                1.045455   
4530                             LET    BLANIK L-13                1.117647   
8809                         GRUMMAN          G164B                0.904762   
4444          EVEKTOR-AEROTECHNIK AS      SPORTSTAR                1.190476   
4374             ROBINSON HELICOPTER            R22                1.238095   
8751                       SCHWEIZER      SGS-2-33A                0.952381   
9082               GRUMMAN-SCHWEIZER         G-164A                0.720000   
4620                           PIPER          PA 18                1.037037   
9311  GRUMMAN ACFT ENG COR-SCHWEIZER         G-164B                0.625000   
4091                           PIPER          PA 28                1.434783   
9294   CONSOLIDATED AERONAUTICS INC.  LAKE LA-4-200                0.640000   
355                           BOEING            747               59.200000   
4586                       SCHWEIZER      SGS 2-33A                1.071429   
4493                           BEECH           C-23                1.147059   
4276                           PIPER      PA-18-160                1.290323   
4061                         ENSTROM          F-28C                1.490196   
3714                           PIPER      PA 28-161                1.750000   
4385                          CESSNA           195B                1.222222   
4071                         ENSTROM          F-28A                1.454545   
4108                           MAULE       M-4-220C                1.421053   
4498                          HILLER         UH-12C                1.142857   
4171              BRANTLY HELICOPTER           B-2B                1.350000   
4303                          BOEING            E75                1.270270   
8994              EAGLE AIRCRAFT CO.           DW-1                0.766667   
4268                        CHAMPION           7GCB                1.300000   
9320                       WEATHERLY           201B                0.620690   
7971                           PIPER      PA-28-150                1.000000   
4152                           PIPER          PA 12                1.378378   
4178                          BOEING    A75N1(PT17)                1.343750   
8895                            BELL         47G-2A                0.833333   
4511                          CESSNA          152II                1.131579   
4434                         AERONCA           15AC                1.200000   
3810                            BELL         206-L4                1.608696   
8993                        ROCKWELL            S2R                0.767442   
8750                           BEECH         A23-19                0.956522   
4501                          CESSNA            305                1.142857   
8854              FLIGHT DESIGN GMBH           CTSW                0.869565   
9073                 AIR TRACTOR INC        AT-502B                0.730769   
9107                 AIR TRACTOR INC         AT-301                0.703704   
9319          ROCKWELL INTERNATIONAL           S-2R                0.625000   
9047                       WEATHERLY           201C                0.750000   

      per.A.Minor.Injuries  per.A.Serious.Injuries  per.Accident.Deaths  \
251               0.242105                0.000000             0.000000   
4162              0.136364                0.000000             0.000000   
4613              0.090909                0.000000             0.000000   
4530              0.176471                0.000000             0.000000   
8809              0.095238                0.000000             0.000000   
4444              0.095238                0.000000             0.000000   
4374              0.047619    

In [205]:
filtered_survival.shape[0]

43

In [206]:
filtered_survival['Make'].value_counts()['BOEING']

4

In [207]:
filtered_survival['Make'].value_counts()

PIPER                             6
BOEING                            4
CESSNA                            3
WEATHERLY                         2
SCHWEIZER                         2
HILLER                            2
AIR TRACTOR INC                   2
ENSTROM                           2
BELL                              2
BEECH                             2
LET                               1
GRUMMAN                           1
MAULE                             1
AVIAT AIRCRAFT INC                1
ROCKWELL INTERNATIONAL            1
ROCKWELL                          1
GRUMMAN ACFT ENG COR-SCHWEIZER    1
BRANTLY HELICOPTER                1
GRUMMAN-SCHWEIZER                 1
EAGLE AIRCRAFT CO.                1
CHAMPION                          1
CONSOLIDATED AERONAUTICS INC.     1
AERONCA                           1
ROBINSON HELICOPTER               1
EVEKTOR-AEROTECHNIK AS            1
FLIGHT DESIGN GMBH                1
Name: Make, dtype: int64

In [208]:
boeing_models = filtered_survival.loc[filtered_survival['Make'] == 'BOEING']
boeing_models

Make        Model  per.Accident.Uninjured  per.A.Minor.Injuries  \
251   BOEING          777               99.147368              0.242105   
355   BOEING          747               59.200000              0.117647   
4178  BOEING  A75N1(PT17)                1.343750              0.250000   
4303  BOEING          E75                1.270270              0.270270   

      per.A.Serious.Injuries  per.Accident.Deaths  Aircraft.Accidents  \
251                 0.000000             0.000000                  23   
355                 0.070588             0.070588                  20   
4178                0.062500             0.062500                  32   
4303                0.054054             0.054054                  37   

      Aircraft.Observations  per.Accident.Severity  Total.Severity  \
251                      95               0.242105        5.568421   
355                      85               0.682353       13.647059   
4178                     32               0.750000       24.000000   
4303                     37               0.702703       26.000000   

      Total.Aircraft.Deaths  Accident.Rate  Mortality.Rate  Survival.Rate  
251                0.000000       0.242105        0.000000       1.000000  
355                1.411765       0.235294        0.001187       0.998813  
4178               2.000000       1.000000        0.036364       0.963636  
4303               2.000000       1.000000        0.032787       0.967213

creating an accident_rate df that filters the merged df for aircraft that meet a threshold number of observations 

In [213]:
safe_flights = merged_frame.loc[
    (merged_frame['Aircraft.Observations'] >= 200)
]

safe_flights.sort_values('Accident.Rate', ascending=True)

Make       Model  per.Accident.Uninjured  per.A.Minor.Injuries  \
374   BOEING         737               52.026585              0.186094   
1751   PIPER   PA-31-350                2.125000              0.317857   
3850   PIPER  PA-34-200T                1.570093              0.247664   
4377   BEECH          58                1.235537              0.115702   
4147  CESSNA       172RG                1.387755              0.195918   
...      ...         ...                     ...                   ...   
4481  CESSNA         177                1.155405              0.378378   
4631   PIPER   PA-18-150                1.022767              0.126095   
4628  CESSNA         140                1.029925              0.182045   
8875  CESSNA        150J                0.853982              0.300885   
4298  CESSNA        182Q                1.274436              0.218045   

      per.A.Serious.Injuries  per.Accident.Deaths  Aircraft.Accidents  \
374                 2.756646             2.756646                 135   
1751                0.846429             0.846429                 259   
3850                0.682243             0.682243                 204   
4377                0.896694             0.896694                 231   
4147                0.383673             0.383673                 237   
...                      ...                  ...                 ...   
4481                0.327703             0.327703                 296   
4631                0.187391             0.187391                 571   
4628                0.114713             0.114713                 401   
8875                0.234513             0.234513                 226   
4298                0.462406             0.462406                 266   

      Aircraft.Observations  per.Accident.Severity  Total.Severity  \
374                     489              22.239264     3002.300613   
1751                    280               7.089286     1836.125000   
3850                    214               5.705607     1163.943925   
4377                    242               7.289256     1683.818182   
4147                    245               3.265306      773.877551   
...                     ...                    ...             ...   
4481                    296               3.000000      888.000000   
4631                    571               1.625219      928.000000   
4628                    401               1.099751      441.000000   
8875                    226               2.176991      492.000000   
4298                    266               3.917293     1042.000000   

      Total.Aircraft.Deaths  Accident.Rate  Mortality.Rate  Survival.Rate  
374              372.147239       0.276074        0.047754       0.952246  
1751             219.225000       0.925000        0.204663       0.795337  
3850             139.177570       0.953271        0.214391       0.785609  
4377             207.136364       0.954545        0.285151       0.714849  
4147              90.930612       0.967347        0.163194       0.836806  
...                     ...            ...             ...            ...  
4481              97.000000       1.000000        0.149691       0.850309  
4631             107.000000       1.000000        0.122989       0.877011  
4628              46.000000       1.000000        0.079585       0.920415  
8875              53.000000       1.000000        0.144414       0.855586  
4298             123.000000       1.000000        0.191291       0.808709  

[62 rows x 14 columns]